crates/replace/src/lib.rs

```diff
@@ -295,72 +296,69 @@ impl<T: Config> Pallet<T> {
         // check vault is not banned
         ext::vault_registry::ensure_not_banned::<T>(&vault_id)?;
 
+        let amount_btc = Amount::new(amount_btc, T::GetWrappedCurrencyId::get());
+        let griefing_collateral = Amount::new(griefing_collateral, T::GetGriefingCollateralCurrencyId::get());
+
         ensure!(
             !ext::nomination::is_nominatable::<T>(&vault_id)?,
             Error::<T>::VaultHasEnabledNomination
         );
 
         let requestable_tokens = ext::vault_registry::requestable_to_be_replaced_tokens::<T>(&vault_id)?;
-        let to_be_replaced_increase = amount_btc.min(requestable_tokens);
+        let to_be_replaced_increase = amount_btc.min(&requestable_tokens)?;
         let replace_collateral_increase = if amount_btc.is_zero() {
             griefing_collateral
         } else {
-            ext::vault_registry::calculate_collateral::<T>(griefing_collateral, to_be_replaced_increase, amount_btc)?
+            ext::vault_registry::calculate_collateral::<T>(&griefing_collateral, &to_be_replaced_increase, &amount_btc)?
         };
 
         // increase to-be-replaced tokens. This will fail if the vault does not have enough tokens available
         let (total_to_be_replaced, total_griefing_collateral) =
             ext::vault_registry::try_increase_to_be_replaced_tokens::<T>(
                 &vault_id,
-                to_be_replaced_increase,
-                replace_collateral_increase,
+                &to_be_replaced_increase,
+                &replace_collateral_increase,
             )?;
 
         // check that total-to-be-replaced is above the minimum. NOTE: this means that even
         // a request with amount=0 is valid, as long the _total_ is above DUST. This might
         // be the case if the vault just wants to increase the griefing collateral, for example.
-        let dust_value = <ReplaceBtcDustValue<T>>::get();
-        ensure!(total_to_be_replaced >= dust_value, Error::<T>::AmountBelowDustAmount);
+        ensure!(
+            total_to_be_replaced.ge(&Self::dust_value())?,
```

Comment:
Feels weird to me that this operation is fallible, i.e. the dust value should always have the same `CurrencyId` but we now have a runtime error if for some reason these are out-of-sync.

Reply:
Well, we also use checked math even when we are sure that the math can not fail, this is not much different. We could compare the `.amount()` values directly, but I think that we should probably keep amount comparisons consistent across the code. Maybe in the future we can extend the type to be able to statically distinguish `Wrapped` and `Collateral` amounts, but I don't think it's very urgent.